### DISCUSSION TITLE
how to always start new TraceId to avoid attack?

### DISCUSSION THREAD

**Discussion by yizhixiaotaozi** [open-telemetry/opentelemetry-java-instrumentation#9251] [Q&A] [answered]
I have a java application which is started with opentelemetry-agent.jar and is a http gateway application. If some one always send requests with a same traceId, then there is a risk to backend storage like clickhouse or es, too many spans with the same traceId!

So I want to start a new span with new traceId every request, ignoring traceparent from outside requests. But span is immutable, I find no api to do this.

Anyone help?

**Comment by trask** [comment, marked as answer]
(transferring to java-instrumentation repo)

you could register your own context propagator which delegates to w3c context propagator for injection (into outgoing requests) and does nothing for extraction (from incoming requests)

in the java agent this would require creating an [extension](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/examples/extension#extensions)

**Reply by yizhixiaotaozi** [reply]
thank you。
